src/cpp/rtps/builtin/data/ParticipantProxyData.cpp

```diff
@@ -455,6 +470,32 @@ bool ParticipantProxyData::readFromCDRMessage(
                         is_shm_transport_available &= (m_VendorId == c_VendorId_eProsima);
                         break;
                     }
+                    case fastdds::dds::PID_PRODUCT_VERSION:
+                    {
+                        VendorId_t local_vendor_id = source_vendor_id;
+                        if (c_VendorId_Unknown == local_vendor_id)
+                        {
+                            local_vendor_id = ((c_VendorId_Unknown == m_VendorId) ? c_VendorId_eProsima : m_VendorId);
+                        }
+
+                        // Ignore custom PID when coming from other vendors
+                        if (c_VendorId_eProsima != local_vendor_id)
+                        {
+                            EPROSIMA_LOG_INFO(RTPS_PROXY_DATA,
+                                    "Ignoring custom PID" << pid << " from vendor " << local_vendor_id);
+                            return true;
+                        }
```

Comment:
Done in 6a78844fa

Reply:
I was thinking more of a `bool is_from_eprosima_vendor` , but this also works.

Reply:
Refactor in  eb1a07

Reply:
> Furthermore, I'm not sure the logic is 100% correct, is it possible that `source_vendor_id` has unknown value? If so, I guess it could be problematic.

This (possibly flawed) logic is present also in writer/reader proxy datas, to be revisited in a future PR.